src/plugins/plugin-http.js

```diff
@@ -0,0 +1,52 @@
+'use strict';
+var shimmer = require('shimmer');
+var url = require('url');
+var isString = require('is').string;
+var merge = require('lodash.merge');
+var httpAgent = require('_http_agent');
+
+function patchModuleRoot (request, api) {
+  var labels = api.labels;
+  function wrapInvocation (options, callback) {
+    return function httpActionTrace () {
+      if (!this._google_trace_patched && options) {
+        // Don't keep wrapping our same request
+        this._google_trace_patched = true;
+        return patchHTTPRequest(options, callback);
+      }
+      return request.apply(this, arguments);
+    }
+  }
+
+  function parseRequestOptions (requestOptions) {
```

Comment:
I don't believe this needs to be in the closure as it doesn't need access to anything held by the closure. Can you pull this out of patchModuleRoot?